### PR TITLE
Fix Rust 1.57 clippy errors

### PR DESCRIPTION
**Pull request by bgurney-rh**
Related: stratis-storage/project#368

### CODE REVIEW

**Comment by bgurney-rh** [comment]
Not quite done yet; still 5 redundant closures to fix, and the fix in the last commit might not be correct.

**Comment by bgurney-rh** [comment]
I have only one clippy error left:
```
error: called `expect` on `max_size` after checking its variant with `is_none`
   --> src/engine/strat_engine/tests/real.rs:118:28
    |
115 |         if max_size.is_none() {
    |         --------------------- help: try: `if let Some(..) = max_size`
...
118 |             let max_size = max_size.expect("!max_size.is_none()");
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
```

...but I can never understand the rust-lang tutorial on "if let"; I think it might need more insight into what variable to put in there.

**Comment by mulkieran** [comment]
> I have only one clippy error left:
> 
> ```
> error: called `expect` on `max_size` after checking its variant with `is_none`
>    --> src/engine/strat_engine/tests/real.rs:118:28
>     |
> 115 |         if max_size.is_none() {
>     |         --------------------- help: try: `if let Some(..) = max_size`
> ...
> 118 |             let max_size = max_size.expect("!max_size.is_none()");
>     |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>     |
> ```
> 
> ...but I can never understand the rust-lang tutorial on "if let"; I think it might need more insight into what variable to put in there.

They want you to switch around a bunch of things to get this statement:
```
    // Retain only devices that are less than max_size
    let (mut matches, mut too_large): (Vec<(&Path, Sectors)>, Vec<(&Path, Sectors)>) =
        if let Some(max_size) = max_size {
            matches.iter().partition(|&&(_, s)| s <= max_size)
        } else {
            (matches, vec![])
        };
```

**Comment by bgurney-rh** [comment]
Okay; it was changing from `if ...is_none()` to `if let Some(...)`.

This latest set of commits has no clippy errors, and builds cleanly in 1.57.0 and 1.56.0.

File: src/engine/strat_engine/liminal/identify.rs [hunk 1]
```diff
@@ -591,6 +591,7 @@ mod tests {
         )
         .unwrap();
 
+        #[allow(clippy::redundant_closure)]
```
Comment (mulkieran) [inline src/engine/strat_engine/liminal/identify.rs:594]: Plz remove any of the redundant closures that you can remove in this file.

**Review by mulkieran — CHANGES_REQUESTED** [review]
One request...

File: src/engine/strat_engine/liminal/identify.rs (hunk 1 above)
Comment (bgurney-rh) [reply src/engine/strat_engine/liminal/identify.rs:594]: I just cherry-picked the commit from when I tried that the last time... and it worked; hooray.

Comment (mulkieran) [reply src/engine/strat_engine/liminal/identify.rs:594]: Great, keep going and remove all allows that you can remove, I _think_ it should be all.

Comment (bgurney-rh) [reply src/engine/strat_engine/liminal/identify.rs]: Ah yes, I have to remove the allow comments now.

Comment (bgurney-rh) [reply src/engine/strat_engine/liminal/identify.rs]: I think this is as far as I can go.  The only ones left are the dbus_api ones, that show "expected signature" errors if I try to remove the allow.

**Review by mulkieran — APPROVED** [review]
Looks good. Plz squash into something reasonable, and I'll merge.

**Pull request closed without merging**